api/client/client.go

```diff
@@ -3389,3 +3389,66 @@ func (c *Client) GetHeadlessAuthentication(ctx context.Context, id string) (*typ
 	}
 	return headlessAuthn, nil
 }
+
+// CreatePlugins creates a new plugin resource
+func (c *Client) CreatePlugin(ctx context.Context, plugin types.Plugin) error {
+	p, ok := plugin.(*types.PluginV1)
+	if !ok {
+		return trace.BadParameter("unknown plugin type")
+	}
+	_, err := c.grpc.CreatePlugin(ctx, p, c.callOpts...)
+	if err != nil {
+		return trail.FromGRPC(err)
+	}
+	return nil
+}
+
+// DeleteAllPlugins deletes all plugin resources
+func (c *Client) DeleteAllPlugins(ctx context.Context) error {
+	_, err := c.grpc.DeleteAllPlugins(ctx, &emptypb.Empty{})
+	return trace.Wrap(err)
+}
+
+// DeletePlugin deletes a plugin resource
+func (c *Client) DeletePlugin(ctx context.Context, name string) error {
+	_, err := c.grpc.DeletePlugin(ctx, &types.ResourceRequest{Name: name})
+	return trace.Wrap(err)
+}
+
+// GetPlugin gets a plugin resource
+func (c *Client) GetPlugin(ctx context.Context, name string, withSecrets bool) (types.Plugin, error) {
+	return c.grpc.GetPlugin(ctx, &types.ResourceRequest{Name: name})
+}
+
+// GetPlugins gets all plugin resources
+func (c *Client) GetPlugins(ctx context.Context, withSecrets bool) (*types.PluginListV1, error) {
+	return c.grpc.GetPlugins(ctx, &emptypb.Empty{})
+}
+
+// ListPlugins returns a list of plugin resources
+func (c *Client) ListPlugins(ctx context.Context, limit int, startKey string, withSecrets bool) ([]types.Plugin, string, error) {
+	request := &types.ListPluginsRequest{
+		Limit:    int32(limit),
+		StartKey: startKey,
+	}
+
+	response, err := c.grpc.ListPlugins(ctx, request, c.callOpts...)
+	if err != nil {
+		return nil, "", trail.FromGRPC(err)
+	}
+
+	plugins := make([]types.Plugin, 0, len(response.Items))
+	for _, plugin := range response.Items {
+		plugins = append(plugins, plugin)
+	}
```

Comment:
Do we really need to create a new list of plugins? Or can we just return the one we got from `response` ?

Reply:
The ones in the response are PluginV1s instead of Plugins.